test_util/test_launch.py

```diff
@@ -0,0 +1,302 @@
+    ParameterValue: 0.0.0.0/0
+  - ParameterKey: PublicSlaveInstanceCount
+    ParameterValue: 1
+  - ParameterKey: SlaveInstanceCount
+    ParameterValue: 5
+""".format(mock_ssh_key_path)
+
+
+def test_aws_cf_simple(capsys, tmpdir, mock_aws_cf_simple):
+    """Test that required parameters are consumed and appropriate output is generated
+    """
+    info, desc = check_success(capsys, tmpdir, mock_aws_cf_simple)
+    # check AWS specific info
+    assert info['type'] == 'cloudformation'
+    assert 'stack_name' in info
+    assert 'region' in info['provider']
+    assert 'access_key_id' in info['provider']
+    assert 'secret_access_key' in info['provider']
+    assert info['ssh']['key_name'] == 'default'
+    assert info['ssh']['delete_with_stack'] is False
+    assert info['ssh']['private_key'] == MOCK_SSH_KEY_DATA
+    assert info['ssh']['user'] == 'core'
+    # check that description is updated with info
+    assert 'stack_name' in desc
+
+
+def test_aws_cf_simple_make_key(capsys, tmpdir, mock_aws_cf_simple):
+    """Test that required parameters are consumed and appropriate output is generated
+    """
+    config = """
+---
+this_is_a_temporary_config_format_do_not_put_in_production: yes_i_agree
+type: cloudformation
+template_url: http://us-west-2.amazonaws.com/downloads
+stack_name: foobar
+provider_info:
+  region: us-west-2
+  access_key_id: asdf09iasdf3m19238jowsfn
+  secret_access_key: asdf0asafawwa3j8ajn
+ssh_info:
+  user: core
+parameters:
+  - ParameterKey: AdminLocation
+    ParameterValue: 0.0.0.0/0
+  - ParameterKey: PublicSlaveInstanceCount
+    ParameterValue: 1
+  - ParameterKey: SlaveInstanceCount
+    ParameterValue: 5
+"""
+    info, desc = check_success(capsys, tmpdir, config)
+    # check AWS specific info
+    assert info['type'] == 'cloudformation'
+    assert 'stack_name' in info
+    assert 'region' in info['provider']
+    assert 'access_key_id' in info['provider']
+    assert 'secret_access_key' in info['provider']
+    assert info['ssh']['key_name'] == 'foobar'
+    assert info['ssh']['delete_with_stack'] is True
+    assert info['ssh']['private_key'] == MOCK_SSH_KEY_DATA
+    assert info['ssh']['user'] == 'core'
+    # check that description is updated with info
+    assert 'stack_name' in desc
+
+
+def test_no_files_specified(tmpdir, mock_aws_cf_simple):
+    """Ensure typical usage works without specifying config and info file paths
+    """
+    with tmpdir.as_cwd():
+        config_path = tmpdir.join('config.yaml')
+        config_path.write(mock_aws_cf_simple)
+        assert main(['create']) == 0
+        assert main(['wait']) == 0
+        assert main(['describe']) == 0
+        assert main(['test']) == 0
+        assert main(['delete']) == 0
+
+
+def test_noop():
+    """Ensure docopt exit (displays usage)
+    """
+    with pytest.raises(DocoptExit):
+        main([])
+    with pytest.raises(DocoptExit):
+        main(['foobar'])
+
+
+def test_conflicting_path(tmpdir, mock_aws_cf_simple):
```

Comment:
Observation: This is very costly test for testing a 'conflicting path` constraint.  Ideally, no cluster should be created for testing this constraint, but in this style, we are creating one cluster for the first `(main['create'])`

(Is it a good idea to refactor code to test these 'constrains of dcos-launch` without actually creating a cluster. If you think is good idea, perhaps this can be revisited with a TODO.